toy_2d/src/ilqr_2d.py

```diff
@@ -0,0 +1,418 @@
+
+    # Simulation parameters.
+    DT = 0.001          # If a generated trajectory looks messed up, it could be
+                        # fixed by making this timestep smaller.
+
+    # Create a polytope.
+    poly_params = TwoDimensionalPolytopeParams(
+        mass = MASS,
+        moment_inertia = MOM_INERTIA,
+        mu_ground = MU_GROUND,
+        vertex_locations = SQUARE_CORNERS
+    )
+    polytope = TwoDimensionalPolytope(poly_params)
+
+    # Create a system from the polytope, a simulation timestep, and a control
+    # contact's friction parameter.
+    system_params = TwoDimensionalSystemParams(
+        dt = DT,
+        polytope = polytope,
+        mu_control = MU_CONTROL
+    )
+
+    # Contact location and direction.
+    CONTACT_LOC = np.array([-1, 1])
+    CONTACT_ANGLE = 0.
+    system = TwoDSystemForceOnly(system_params, CONTACT_LOC, CONTACT_ANGLE)
+    # Create an LCS approximation from the system.
+    lcs = TwoDSystemLCSApproximation(system)
+
+    #Following the hw methodology of implementing ILQR
+    #take the current state and action and give the current cost. Here the cost can be 
+    #non-linear but defining a quadratic cost makes life easier
+    num_states = 6
+    num_controls = 2
+    # Initial conditions, in order of x, dx, y, dy, theta, dtheta
+    x0 = np.array([0, 0, 1.0, 0, 0, 0])
+    #goal to be achieved
+    x_goal = np.array([5.0, 0, 1.0, 0, np.pi/2, 0])
+    #traj penalty which has to be tuned depending on where the goal is 
+    Q = np.eye(num_states)
+    # Q[1,1]=Q[1,1]*0
+    # Q[3,3]=Q[3,3]*0
+    # Q[5,5]=Q[5,5]*0
+    #final goal penalty. Needs tuning
+    Qf = np.eye(num_states)*1000
+    #penalty on using more force. Experience suggests making this zero has bad effects
+    R = np.eye(num_controls)*0.05
+    #number of timesteps in gthe rollout
+    num_timesteps = 8000
+    #initial guess required the most tuning. Experience suggests making this high initially  
+    #then let it optimize and cool down to the necessary value. Since we are putting a penalty on 
+    #the amount of force, starting with a low value might not be the best thing to do. Remember this is 
+    #a nonlinear program and if you start with a bad guess, you wont get a good solution.
+    u_guess = np.ones((num_timesteps-1,num_controls))*2
+    #load your previous run answer to get a warm start. Remember, if you change num_timesteps you will
+    #get error loading the previous solution, the timesteps also decide the num of control steps. 
+    #If you run the script with the same num_timesteps (as a previous run) and want to try different cost
+    # make he flag true 
+    load_old_run = False  
+    if(load_old_run):
+        file_name = "controls"
+        path = f'{file_utils.OUT_DIR}/{file_name}.txt'
+        with open (path, 'rb') as fp:
+            u_guess = pickle.load(fp)
+
+    obj = iLQR(x0,x_goal,num_timesteps,DT,num_states,num_controls,Q,R,Qf)
```

Comment:
Many of these arguments to initialize an `iLQR` instance are unnecessary, given many of these parameters are embedded in the `lcs` object referenced in the `iLQR` class.  I suggest the following as necessary inputs to the `iLQR` class:
- `lcs`
- `x_goal`
- `Q`
- `R`
- `Qf`

Then the rest of the current inputs can be obtained via:
- Using `lcs.set_initial_state(x0)` means `x0` can be accessed via `lcs.state_history[0]` (note:  LCS versus state format matters here, i.e. use the correct version of `[vx, vy, vth, x, y, th]` or `[x, vx, y, vy, th, vth]`.)
- `lcs.system.params.dt` yields `DT`.
- `lcs.state_history.shape[1]` or `Q.shape[0]` yield `num_states`.
- `lcs.get_P_matrix().shape[1]` or `R.shape[0]` yield `num_controls`.

Reply:
These are great suggestions thanks. Since, I was prototyping, I made this hasty mistake of using external objects. I ended up using the param inputs you suggested and one more param which is num_timesteps.